tools/shared.py

```diff
@@ -530,15 +530,19 @@ def check_node_version():
 
 
 def check_closure_compiler():
-  try:
-    run_process([JAVA, '-version'], stdout=PIPE, stderr=PIPE)
-  except Exception:
-    warning('java does not seem to exist, required for closure compiler, which is optional (define JAVA in ' + hint_config_file_location() + ' if you want it)')
-    return False
   if not os.path.exists(CLOSURE_COMPILER):
-    warning('Closure compiler (%s) does not exist, check the paths in %s', CLOSURE_COMPILER, EM_CONFIG)
+    warning('Closure compiler (%s) does not exist, check the paths in %s. To install Closure compiler, run "npm install" in Emscripten root directory.', CLOSURE_COMPILER, EM_CONFIG)
     return False
-  return True
+  try:
+    proc = subprocess.Popen([CLOSURE_COMPILER, '--version'], stdout=PIPE, stderr=PIPE)
+    output = proc.communicate()
+    if 'Version:' in str(output[0]):
+      return True
+    warning('Unrecognized Closure compiler --version output:\n' + str(output[0]) + '\n' + str(output[1]))
+  except Exception as e:
+    warning('Closure compiler ("%s --version") did not execute properly!' % CLOSURE_COMPILER)
+    warning(str(e))
+  return False
```

Comment:
I think maybe we should switch this over to only checking on demand, rather than on startup/sanity, but thats a separate change.